requirements.txt

```diff
@@ -5,3 +5,5 @@ tensorboard
 sentencepiece
 tiktoken
 blobfile
+-i https://download.pytorch.org/whl/nightly/cpu
```

Comment:
What is the earliest nightly that is good to test? Apr 25?

We depend on nightly pytorch in CI, so this can be removed when we land this PR.
https://github.com/pytorch/torchtitan/blob/main/.github/workflows/unit_test_4gpu.yaml#L36

Reply:
@tianyu-l Need torchdata build from Apr 26 (20240426) at least

Reply:
for HF datasets it seems from local test that we need `datasets >= 2.19.0` to use the `skip()` API

Reply:
confirm that I need to reinstall datasets to run with the PR